service/controller/v12/resource/instance/error.go

```diff
@@ -99,3 +99,19 @@ var versionBlobEmptyError = &microerror.Error{
 func IsVersionBlobEmpty(err error) bool {
 	return microerror.Cause(err) == versionBlobEmptyError
 }
+
+var nilTemplateLinkError = &microerror.Error{
+	Kind: "nilTemplateLink",
+}
+
+func IsNilTemplateLinkError(err error) bool {
+	return microerror.Cause(err) == nilTemplateLinkError
+}
+
+var unableToGetTemplateError = &microerror.Error{
+	Kind: "unableToGetTemplate",
+}
+
+func IsUnableToGetTemplateError(err error) bool {
```

Comment:
Since we don't use `IsNilTemplateLinkError` & `IsNilTemplateLinkError` we should replace those errors with `executionFailedError`. https://github.com/giantswarm/fmt/blob/master/go/errors.md#matching-errors

Reply:
I am actually using those in unit tests to check that the right error type is returned.
Is it acceptable or should I replace them with the generic error as you suggested?